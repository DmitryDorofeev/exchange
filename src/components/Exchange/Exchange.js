import React from 'react';
import PropTypes from 'prop-types';
import {PocketInfo} from '../PocketInfo/PocketInfo';
import {Slider} from '../Slider/Slider';
import styles from './Exchange.module.css';
import {Button} from '../Button/Button';

export const Exchange = ({pockets, from, to, onExchange, onInputChange, onFromSwitch, onToSwitch}) => {
  return (
    <div className={styles.root}>
      <Slider
        onChange={(index) => onFromSwitch(pockets[index].currency)}
        active={pockets.findIndex((pocket) => pocket.currency === from.currency)}
      >
        {pockets.map((pocket, key) => (
          <PocketInfo
            amount={pocket.amount}
            currency={pocket.currency}
            onInputChange={onInputChange}
            value={from.value}
            editable
            key={key}
          />
        ))}
      </Slider>
      <Slider
        onChange={(index) => onToSwitch(pockets[index].currency)}
        active={pockets.findIndex((pocket) => pocket.currency === to.currency)}
      >
        {pockets.map((pocket, key) => (
          <PocketInfo
            amount={pocket.amount}
            currency={pocket.currency}
            value={to.value}
            key={key}
          />
        ))}
      </Slider>
      <div className={styles.button}>
        <Button
          disabled={from.currency === to.currency || from.value > from.amount}
          onClick={() => onExchange({from, to})}
        >
          Exchange
        </Button>
      </div>
    </div>
  )
};

Exchange.defaultProps = {
  onExchange: () => null,
  onInputChange: () => null,
  onFromSwitch: () => null,
  onToSwitch: () => null,
  pockets: [],
};

const PocketType = PropTypes.shape({
  amount: PropTypes.number.isRequired,
  currency: PropTypes.string.isRequired,
});

Exchange.propTypes = {
  from: PocketType.isRequired,
  to: PocketType.isRequired,
};
