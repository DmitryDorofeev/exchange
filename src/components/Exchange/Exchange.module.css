.root {

}

.button {
    display: flex;
    justify-content: center;
}