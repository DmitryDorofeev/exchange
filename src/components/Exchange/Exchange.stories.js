import React from 'react';
import {storiesOf} from '@storybook/react';
import {Exchange} from './Exchange';

const from = {
  value: 30,
  currency: 'GBP',
};

const to = {
  value: 50,
  currency: 'EUR',
};

const pockets = [
  {amount: 300, currency: 'EUR'},
  {amount: 3000, currency: 'GBP'},
];

storiesOf('Exchange', module)
  .add('default', () => <Exchange pockets={pockets} from={from} to={to}/>);