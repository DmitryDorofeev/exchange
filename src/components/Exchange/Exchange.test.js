import React from 'react';
import {shallow} from 'enzyme';
import {Exchange} from './Exchange';

describe('Exchange component', () => {

  it('should match snapshot', () => {
    const pockets = [
      {amount: 3, currency: 'USD'},
      {amount: 3, currency: 'GBP'},
    ];
    const from = {value: 3, currency: 'USD'};
    const to = {value: 3, currency: 'GBP'};
    const exchange = shallow(<Exchange pockets={pockets} from={from} to={to}/>);

    expect(exchange).toMatchSnapshot();
  });

  it('should be disable when currencies are same', () => {
    const pockets = [
      {amount: 3, currency: 'USD'},
      {amount: 3, currency: 'GBP'},
    ];
    const from = {value: 3, currency: 'USD'};
    const to = {value: 3, currency: 'USD'};
    const exchange = shallow(<Exchange pockets={pockets} from={from} to={to}/>);

    expect(exchange).toMatchSnapshot();
  });

  it('should be disable when value is less then amount', () => {
    const pockets = [
      {value: 3, currency: 'USD'},
      {value: 3, currency: 'GBP'},
    ];
    const from = {value: 4, currency: 'USD'};
    const to = {value: 3, currency: 'GBP'};
    const exchange = shallow(<Exchange pockets={pockets} from={from} to={to}/>);

    expect(exchange).toMatchSnapshot();
  })
});