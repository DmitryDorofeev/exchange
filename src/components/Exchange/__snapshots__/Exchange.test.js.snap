// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange component should be disable when currencies are same 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exchange
    from={
      Object {
        "currency": "USD",
        "value": 3,
      }
    }
    onExchange={[Function]}
    onFromSwitch={[Function]}
    onInputChange={[Function]}
    onToSwitch={[Function]}
    pockets={
      Array [
        Object {
          "amount": 3,
          "currency": "USD",
        },
        Object {
          "amount": 3,
          "currency": "GBP",
        },
      ]
    }
    to={
      Object {
        "currency": "USD",
        "value": 3,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Slider
          active={0}
          onChange={[Function]}
        >
          <PocketInfo
            amount={3}
            currency="USD"
            editable={true}
            onInputChange={[Function]}
            value={3}
          />
          <PocketInfo
            amount={3}
            currency="GBP"
            editable={true}
            onInputChange={[Function]}
            value={3}
          />
        </Slider>,
        <Slider
          active={0}
          onChange={[Function]}
        >
          <PocketInfo
            amount={3}
            currency="USD"
            onInputChange={[Function]}
            value={3}
          />
          <PocketInfo
            amount={3}
            currency="GBP"
            onInputChange={[Function]}
            value={3}
          />
        </Slider>,
        <div
          className="button"
        >
          <Button
            disabled={true}
            onClick={[Function]}
          >
            Exchange
          </Button>
        </div>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 0,
          "children": Array [
            <PocketInfo
              amount={3}
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />,
            <PocketInfo
              amount={3}
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "USD",
              "editable": true,
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "GBP",
              "editable": true,
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 0,
          "children": Array [
            <PocketInfo
              amount={3}
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />,
            <PocketInfo
              amount={3}
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "USD",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "GBP",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            disabled={true}
            onClick={[Function]}
          >
            Exchange
          </Button>,
          "className": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Exchange",
            "disabled": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Exchange",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Slider
            active={0}
            onChange={[Function]}
          >
            <PocketInfo
              amount={3}
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />
            <PocketInfo
              amount={3}
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />
          </Slider>,
          <Slider
            active={0}
            onChange={[Function]}
          >
            <PocketInfo
              amount={3}
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />
            <PocketInfo
              amount={3}
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />
          </Slider>,
          <div
            className="button"
          >
            <Button
              disabled={true}
              onClick={[Function]}
            >
              Exchange
            </Button>
          </div>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 0,
            "children": Array [
              <PocketInfo
                amount={3}
                currency="USD"
                editable={true}
                onInputChange={[Function]}
                value={3}
              />,
              <PocketInfo
                amount={3}
                currency="GBP"
                editable={true}
                onInputChange={[Function]}
                value={3}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "USD",
                "editable": true,
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "GBP",
                "editable": true,
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 0,
            "children": Array [
              <PocketInfo
                amount={3}
                currency="USD"
                onInputChange={[Function]}
                value={3}
              />,
              <PocketInfo
                amount={3}
                currency="GBP"
                onInputChange={[Function]}
                value={3}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "USD",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "GBP",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              disabled={true}
              onClick={[Function]}
            >
              Exchange
            </Button>,
            "className": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Exchange",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Exchange",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Exchange component should be disable when value is less then amount 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exchange
    from={
      Object {
        "currency": "USD",
        "value": 4,
      }
    }
    onExchange={[Function]}
    onFromSwitch={[Function]}
    onInputChange={[Function]}
    onToSwitch={[Function]}
    pockets={
      Array [
        Object {
          "currency": "USD",
          "value": 3,
        },
        Object {
          "currency": "GBP",
          "value": 3,
        },
      ]
    }
    to={
      Object {
        "currency": "GBP",
        "value": 3,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Slider
          active={0}
          onChange={[Function]}
        >
          <PocketInfo
            currency="USD"
            editable={true}
            onInputChange={[Function]}
            value={4}
          />
          <PocketInfo
            currency="GBP"
            editable={true}
            onInputChange={[Function]}
            value={4}
          />
        </Slider>,
        <Slider
          active={1}
          onChange={[Function]}
        >
          <PocketInfo
            currency="USD"
            onInputChange={[Function]}
            value={3}
          />
          <PocketInfo
            currency="GBP"
            onInputChange={[Function]}
            value={3}
          />
        </Slider>,
        <div
          className="button"
        >
          <Button
            disabled={false}
            onClick={[Function]}
          >
            Exchange
          </Button>
        </div>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 0,
          "children": Array [
            <PocketInfo
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={4}
            />,
            <PocketInfo
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={4}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": undefined,
              "currency": "USD",
              "editable": true,
              "onInputChange": [Function],
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": undefined,
              "currency": "GBP",
              "editable": true,
              "onInputChange": [Function],
              "value": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 1,
          "children": Array [
            <PocketInfo
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />,
            <PocketInfo
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": undefined,
              "currency": "USD",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": undefined,
              "currency": "GBP",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            disabled={false}
            onClick={[Function]}
          >
            Exchange
          </Button>,
          "className": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Exchange",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Exchange",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Slider
            active={0}
            onChange={[Function]}
          >
            <PocketInfo
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={4}
            />
            <PocketInfo
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={4}
            />
          </Slider>,
          <Slider
            active={1}
            onChange={[Function]}
          >
            <PocketInfo
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />
            <PocketInfo
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />
          </Slider>,
          <div
            className="button"
          >
            <Button
              disabled={false}
              onClick={[Function]}
            >
              Exchange
            </Button>
          </div>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 0,
            "children": Array [
              <PocketInfo
                currency="USD"
                editable={true}
                onInputChange={[Function]}
                value={4}
              />,
              <PocketInfo
                currency="GBP"
                editable={true}
                onInputChange={[Function]}
                value={4}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": undefined,
                "currency": "USD",
                "editable": true,
                "onInputChange": [Function],
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": undefined,
                "currency": "GBP",
                "editable": true,
                "onInputChange": [Function],
                "value": 4,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 1,
            "children": Array [
              <PocketInfo
                currency="USD"
                onInputChange={[Function]}
                value={3}
              />,
              <PocketInfo
                currency="GBP"
                onInputChange={[Function]}
                value={3}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": undefined,
                "currency": "USD",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": undefined,
                "currency": "GBP",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              disabled={false}
              onClick={[Function]}
            >
              Exchange
            </Button>,
            "className": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Exchange",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Exchange",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Exchange component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Exchange
    from={
      Object {
        "currency": "USD",
        "value": 3,
      }
    }
    onExchange={[Function]}
    onFromSwitch={[Function]}
    onInputChange={[Function]}
    onToSwitch={[Function]}
    pockets={
      Array [
        Object {
          "amount": 3,
          "currency": "USD",
        },
        Object {
          "amount": 3,
          "currency": "GBP",
        },
      ]
    }
    to={
      Object {
        "currency": "GBP",
        "value": 3,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Slider
          active={0}
          onChange={[Function]}
        >
          <PocketInfo
            amount={3}
            currency="USD"
            editable={true}
            onInputChange={[Function]}
            value={3}
          />
          <PocketInfo
            amount={3}
            currency="GBP"
            editable={true}
            onInputChange={[Function]}
            value={3}
          />
        </Slider>,
        <Slider
          active={1}
          onChange={[Function]}
        >
          <PocketInfo
            amount={3}
            currency="USD"
            onInputChange={[Function]}
            value={3}
          />
          <PocketInfo
            amount={3}
            currency="GBP"
            onInputChange={[Function]}
            value={3}
          />
        </Slider>,
        <div
          className="button"
        >
          <Button
            disabled={false}
            onClick={[Function]}
          >
            Exchange
          </Button>
        </div>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 0,
          "children": Array [
            <PocketInfo
              amount={3}
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />,
            <PocketInfo
              amount={3}
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "USD",
              "editable": true,
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "GBP",
              "editable": true,
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": 1,
          "children": Array [
            <PocketInfo
              amount={3}
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />,
            <PocketInfo
              amount={3}
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "USD",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "amount": 3,
              "currency": "GBP",
              "onInputChange": [Function],
              "value": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Button
            disabled={false}
            onClick={[Function]}
          >
            Exchange
          </Button>,
          "className": "button",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Exchange",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Exchange",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Slider
            active={0}
            onChange={[Function]}
          >
            <PocketInfo
              amount={3}
              currency="USD"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />
            <PocketInfo
              amount={3}
              currency="GBP"
              editable={true}
              onInputChange={[Function]}
              value={3}
            />
          </Slider>,
          <Slider
            active={1}
            onChange={[Function]}
          >
            <PocketInfo
              amount={3}
              currency="USD"
              onInputChange={[Function]}
              value={3}
            />
            <PocketInfo
              amount={3}
              currency="GBP"
              onInputChange={[Function]}
              value={3}
            />
          </Slider>,
          <div
            className="button"
          >
            <Button
              disabled={false}
              onClick={[Function]}
            >
              Exchange
            </Button>
          </div>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 0,
            "children": Array [
              <PocketInfo
                amount={3}
                currency="USD"
                editable={true}
                onInputChange={[Function]}
                value={3}
              />,
              <PocketInfo
                amount={3}
                currency="GBP"
                editable={true}
                onInputChange={[Function]}
                value={3}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "USD",
                "editable": true,
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "GBP",
                "editable": true,
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": 1,
            "children": Array [
              <PocketInfo
                amount={3}
                currency="USD"
                onInputChange={[Function]}
                value={3}
              />,
              <PocketInfo
                amount={3}
                currency="GBP"
                onInputChange={[Function]}
                value={3}
              />,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "USD",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "amount": 3,
                "currency": "GBP",
                "onInputChange": [Function],
                "value": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Button
              disabled={false}
              onClick={[Function]}
            >
              Exchange
            </Button>,
            "className": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Exchange",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Exchange",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
